engine/gfx/sprite_palettes.asm

```diff
@@ -22,7 +22,9 @@ LoadWeatherPal::
 	push af
 	ld a, BANK(wOBPals1)
 	ldh [rSVBK], a
-	ld a, -1
+	; we are not loading an official palette, so it's okay
+	; if the dynamic pal system uses this slot later.
+	ld a, NO_PAL_LOADED
 	ld [wLoadedObjPal7], a
```

Comment:
The "heading" variable name is `wUsedObjectPals`, but then the individual members are `wLoadedObjPal0-7`. Can they be in sync? So either rename to `wLoadedObjectPal` or `wUsedObjPal0-7`.

Reply:
Technically they are unrelated. `wUsedObjectPals` is a bit array. Whereas `wLoadedOBjPal0-7` are bytes.  
The former marks which pal slots are in use, the latter marks what palette indexes are loaded in those slots.

Reply:
Oh, whoops! I was confused because I saw `NO_PAL_LOADED` being assigned to `wLoadedObjPal7`, but also filled in the array starting with `wUsedObjectPals`. Was your intention to assign `-1` to `wUsedObjectPals` as well as `wLoadedObjPal0`, `wLoadedObjPal1`, etc?

Reply:
This looks... off to me now that i'm looking at it..  
  
\`\`\`  
    xor a

    ld \[wUsedObjectPals\], a

    ld hl, wUsedObjectPals

    ld bc, wNeededPalIndex - wUsedObjectPals

    ld a, NO\_PAL\_LOADED

    rst ByteFill  
\`\`\`

Reply:
Should be this

```
	xor a
	ld [wUsedObjectPals], a
	ld hl, wLoadedObjPal0
	ld bc, wNeededPalIndex - wLoadedObjPal0
	ld a, NO_PAL_LOADED
	rst ByteFill
```

Reply:
Yeah, that makes more sense. :) Thanks!